src/widgets/InfoBadge.js

```diff
@@ -0,0 +1,129 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { StyleSheet, View, TouchableHighlight, Text } from 'react-native';
+import Popover from 'react-native-popover-view';
+import Badge from './Badge';
+import { getBadgeData } from '../utilities/getBadgeData';
+import { SUSSOL_ORANGE } from '../globalStyles';
+
+export class InfoBadge extends React.PureComponent {
+  state = {
+    isPopOverVisible: false,
+    rect: {},
+  };
+
+  showPopover() {
+    // Open popover and determine where to open it on canvas
+    this.touchableContainer.measure((ox, oy, width, height, px, py) => {
+      this.setState({
+        isPopOverVisible: true,
+        rect: { x: px, y: py, width, height },
+      });
+    });
+  }
+
+  closePopover() {
+    this.setState({ isPopOverVisible: false });
+  }
+
+  render() {
+    const { children, mainWrapperStyle, popoverPosition, routeName } = this.props;
+    const { isPopOverVisible, rect } = this.state;
+    const info = getBadgeData(routeName);
+
+    // Get total of all the count variables in the info array. We want to show it on the badge
+    const pendingCount = info.reduce((total, item) => total + (item.count || 0), 0);
+
+    // Since having multiple digit text can break the badge, we are showing 99+
+    // if the number is greater then 99.
+    const unfinalisedCountText = pendingCount > 99 ? '99+' : pendingCount;
+
+    return (
+      <View style={{ position: 'relative' }}>
+        {children}
+        {pendingCount !== 0 && (
+          <View style={[localStyles.mainWrapperStyle, mainWrapperStyle]}>
+            <TouchableHighlight
+              ref={ref => {
```

Comment:
I think you could put `touchableHighlight` in with the badge component and forward a ref - https://reactjs.org/docs/forwarding-refs.html - not needed